### ISSUE TITLE
[Bug]: Slider handle focus ring on lower z-order than track

### ISSUE THREAD

**Issue by kikiapple** [adobe/spectrum-web-components#3688]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Impacted component(s)

sp-slider

### Expected behavior

The full slider handle focus ring should be visible above the slider track.
![screen_shot_2023-09-29_at_10 12 57_am_720](https://github.com/adobe/spectrum-web-components/assets/53454136/1832da75-addf-48c2-a3b7-801b1f335813)

### Actual behavior

A portion of the slider handle focus ring is hidden behind the slider track.
<img width="214" alt="Screen Shot 2023-09-29 at 10 19 43 AM" src="https://github.com/adobe/spectrum-web-components/assets/53454136/7cd36c7f-7c93-4c49-b0ec-33621f29e58d">

### What browsers are you seeing the problem in?

Chrome

### How can we reproduce this issue?

1. Go to [Adobe Express](https://new.express.adobe.com/)
2. Create a new project.
3. Add media to the project and select the media.
4. Find any slider in the Properties Inspector.
5. Click and drag the slider handle, press the Cmd key for the focus ring to appear.
6. Observe that the focus ring gets hidden by the track on the right side of the handle.
